Unable to build native image

Hi guys! I am having this error while building native image:
Error: substitution target for io.quarkiverse.googlecloudservices.common.runtime.graal.DefaultHttpTransportFactorySubstitution is not loaded. Use field `onlyWith` in the `TargetClass` annotation to make substitution only active when needed.
com.oracle.svm.core.util.UserError$UserException: substitution target for io.quarkiverse.googlecloudservices.common.runtime.graal.DefaultHttpTransportFactorySubstitution is not loaded. Use field `onlyWith` in the `TargetClass` annotation to make substitution only active when needed.
	at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.findTargetClass(AnnotationSubstitutionProcessor.java:834)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:252)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:230)
	at com.oracle.svm.hosted.NativeImageGenerator.createDeclarativeSubstitutionProcessor(NativeImageGenerator.java:908)
	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:838)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:536)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:451)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------


I am using graalvm Running Quarkus native-image plugin on GraalVM Version 20.1.0 (Java Version 11.0.7), trying to compile the example Pubsub at https://github.com/quarkiverse/quarkiverse-google-cloud-services/tree/master/pubsub
quarkus 1.9.1.Final

thanks in advance

I am not lucky with older versions. GraalVM upgrade is a must.

[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.IllegalStateException: Out of date version of GraalVM detected: GraalVM Version 19.3.2 CE. Quarkus currently supports 20.2. Please upgrade GraalVM to this version.
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.checkGraalVMVersion(NativeImageBuildStep.java:528)

yes I build in by source code directly , because building with :
mvn io.quarkus:quarkus-maven-plugin:1.8.3.Final:create -DprojectGroupId=org.acme  -DprojectArtifactId=pubsub-quickstart-dpp  -Dextensions="quarkus-google-cloud-pubsub"  
throw me an error 
Cannot find a dependency matching 'quarkus-google-cloud-pubsub', maybe a typo?
[ERROR] Unable to install the Maven wrapper (./mvnw) in the project
org.apache.maven.project.ProjectBuildingException: 1 problem was encountered while building the effective model
[FATAL] Non-readable POM ..... (No found file) 

Lets to try with 0.1.2 , but the same issue "...Error: substitution target..."

I updated the repo if you want to reproduce the behaviour

--- a/common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/runtime/graal/DefaultHttpTransportFactorySubstitution.java
+++ b/common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/runtime/graal/DefaultHttpTransportFactorySubstitution.java
@@ -1,16 +1,31 @@
 package io.quarkiverse.googlecloudservices.common.runtime.graal;
 
+import java.util.function.BooleanSupplier;
+
 import com.google.api.client.http.HttpTransport;
 import com.google.api.client.http.javanet.NetHttpTransport;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
 
-@TargetClass(className = "com.google.cloud.http.HttpTransportOptions$DefaultHttpTransportFactory")
+@TargetClass(className = "com.google.cloud.http.HttpTransportOptions$DefaultHttpTransportFactory", onlyWith = DefaultHttpTransportFactorySubstitution.OnlyIfInClassPath.class)
 public final class DefaultHttpTransportFactorySubstitution {
     @Substitute
     public HttpTransport create() {
         // Appengine HttpTransport didn't works on native image.
         // Anyway, appengine don't allow to deploy native image on it so it's not an issue.
         return new NetHttpTransport();
+    }
+
+    public static class OnlyIfInClassPath implements BooleanSupplier {
+        @Override
+        public boolean getAsBoolean() {
+            try {
+                Class.forName("com.google.cloud.http.HttpTransportOptions$DefaultHttpTransportFactory", false,
+                        Thread.currentThread().getContextClassLoader());
+                return true;
+            } catch (ClassNotFoundException e) {
+                return false;
+            }
+        }
     }
 }

--- a/integration-tests/app-engine/pom.xml
+++ b/integration-tests/app-engine/pom.xml
@@ -10,12 +10,11 @@
   </parent>
 
   <artifactId>quarkus-google-cloud-services-app-engine-it</artifactId>
-  <name>Quarkus - Google Cloud Services - Integration Tests - Functions</name>
+  <name>Quarkus - Google Cloud Services - Integration Tests - AppEngine</name>
 
   <properties>
     <maven.deploy.skip>true</maven.deploy.skip>
     <gpg.skip>true</gpg.skip>
-    <quarkus.build.skip>${skipTests}</quarkus.build.skip>
     <native.surefire.skip>${skipTests}</native.surefire.skip>
   </properties>
 

--- a/integration-tests/google-cloud-functions/pom.xml
+++ b/integration-tests/google-cloud-functions/pom.xml
@@ -16,7 +16,6 @@
     <properties>
         <maven.deploy.skip>true</maven.deploy.skip>
         <gpg.skip>true</gpg.skip>
-        <quarkus.build.skip>${skipTests}</quarkus.build.skip>
         <native.surefire.skip>${skipTests}</native.surefire.skip>
     </properties>
 

--- a/integration-tests/main/pom.xml
+++ b/integration-tests/main/pom.xml
@@ -15,7 +15,6 @@
   <properties>
     <maven.deploy.skip>true</maven.deploy.skip>
     <gpg.skip>true</gpg.skip>
-    <quarkus.build.skip>${skipTests}</quarkus.build.skip>
     <native.surefire.skip>${skipTests}</native.surefire.skip>
   </properties>
 

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -17,7 +17,6 @@
     <properties>
         <maven.deploy.skip>true</maven.deploy.skip>
         <gpg.skip>true</gpg.skip>
-        <quarkus.build.skip>${skipTests}</quarkus.build.skip>
         <native.surefire.skip>${skipTests}</native.surefire.skip>
     </properties>
 

